### PR TITLE
Avoid erasing files when Ruff fails with an error

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

A user reported an issue in the VS Code extension (https://github.com/astral-sh/ruff/issues/9123) in which their files were being erased on-save. The issue is that we don't check the exit code of all commands, and so if `ruff check` fails with an _error_, then we end up wiping the file entirely.

This PR adds exit code-checking for all commands in the LSP. Further, we now surface such failures to the user directly (see bottom right):

Screen Shot 2023-12-13 at 10 57 20 PM

Closes https://github.com/astral-sh/ruff/issues/9123

### CODE REVIEW

File: ruff_lsp/server.py [hunk 1]
```diff
@@ -558,6 +558,14 @@ async def _lint_document_impl(document: Document) -> list[Diagnostic]:
     result = await _run_check_on_document(document)
     if result is None:
         return []
+
+    # For `ruff check`, 0 indicates successful completion with no remaining
+    # diagnostics, 1 indicates successful completion with remaining diagnostics,
+    # and 2 indicates an error.
+    if result.exit_code != 0 and result.exit_code != 1:
```
Comment (MichaReiser) [inline ruff_lsp/server.py:565]: Nit: I don't know what's more idiomatic Python
```suggestion
    if not result.exit_code in [0, 1]:
```

File: ruff_lsp/server.py [hunk 2]
```diff
@@ -1141,6 +1159,17 @@ async def _fix_document_impl(
         extra_args=["--fix"],
         only=only,
     )
+
+    if result is None:
```
Comment (MichaReiser) [inline ruff_lsp/server.py:1163]: When does `ruff` return `None`? if there were no fixes? Is this an error state (same for the other commands)

**Review by MichaReiser — APPROVED** [review]

File: ruff_lsp/server.py (hunk 1 above)
Comment (dhruvmanila) [reply ruff_lsp/server.py:565]: 😬 
```python
if result.exit_code not in [0, 1]:
	...
```

File: ruff_lsp/server.py (hunk 2 above)
Comment (charliermarsh) [reply ruff_lsp/server.py:1163]: `_run_check_on_document` will return `None` if the _extension_ chooses to ignore the file (for example, if it appears to be a stdlib file in the current environment).

File: ruff_lsp/server.py (hunk 1 above)
Comment (charliermarsh) [reply ruff_lsp/server.py]: I think `not in (0, 1)` is most idiomatic, probably.

Comment (MichaReiser) [reply ruff_lsp/server.py]: > 😬
> 
> ```python
> if result.exit_code not in [0, 1]:
> 	...
> ```

Python's hard

Comment (charliermarsh) [reply ruff_lsp/server.py]: Haha both work!

**Review by dhruvmanila — APPROVED** [review]
Thanks for the quick fix! It works on Notebook as expected.

**Pull request merged**